docs/source/dialogs.md

```diff
@@ -65,3 +69,23 @@ def process(self):
         ips = [entry[4][0] for entry in resolved if not '%' in entry[4][0]]
         self.produce(ResolvedHostname(name=hostname, ips=ips))
 ```
+
+### Explaining the dialogs processing mechanism during the upgrade
+
+The upgrade itself from the operator's point of view consists of 3 distinct stages: leapp preupgrade, leapp answer
+and leapp upgrade.
```

Comment:
In such case, rather use: `preupgrade, remediation, upgrade`. using `leapp answer` is just part of remediation stage, during which user should do whatever else it is required regarding the report (e.g. create specific file, remove specific rpms, ....).